gems/smithy-json/spec/smithy-json/codec_spec.rb

```diff
@@ -47,60 +43,60 @@ module JSON
           union: { string: 'string' }
         }
         data = data.merge(structure: data)
-        bytes = subject.serialize(structure_shape, data)
-        expect(subject.deserialize(structure_shape, bytes).to_h).to eq(data)
+        json = subject.serialize(structure_shape, data)
+        expect(subject.deserialize(structure_shape, json).to_h).to eq(data)
       end
 
       context 'structures' do
```

Comment:
Unrelated to defaults but we should probably add a test related to timestampFormat trait.
(to ensure that we resolve timestamp formats per [here](https://smithy.io/2.0/spec/protocol-traits.html#timestampformat-trait))

Reply:
This is tested with protocol tests but we can add something extra.